File: hsg9-tinh/2023-2024-cantho/bangso/bangso.py
import os
import sys

input_file = os.path.join(sys.path[0], 'bangso.inp')
output_file = os.path.join(sys.path[0], 'bangso.out')

n = k = 0
count = 0


def input():
    global n, k

    with open(input_file, 'r') as f:
        n, k = map(int, f.readline().split())


def process():
    global n, k, count

    # Duyệt từng chỉ số hàng trong đoạn [1..n]
    for i in range(1, n):
        # Nếu k không chia hết cho i thì không
        if k % i != 0:
            continue

        # Tính giá trị cột j
        j = k // i

        # Kiểm tra xem j có nằm trong phạm vi của bảng không (1 <= j <= n)
        if j <= n:
            count += 1


def output():
    global count

    with open(output_file, "w") as f:
        f.write(str(count))


if __name__ == '__main__':
    input()
    process()
    output()

File: hsg9-tinh/2023-2024-cantho/bangso/bangso.cpp
#include <iostream>

#define input_file "bangso.inp"
#define output_file "bangso.out"

using namespace std;

typedef long long int lli;

lli n, k;
lli count = 0;


void input()
{
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);

    freopen(input_file, "r", stdin);

    cin >> n >> k;
}


void process()
{
    // Duyệt từng chỉ số hàng trong đoạn [1..n]
    for (lli i = 1; i < n + 1; ++i)
    {
        // Nếu k không chia hết cho i thì không
        if (k % i != 0) continue;

        // Tính giá trị cột j
        lli j = k / i;

        // Kiểm tra xem j có nằm trong phạm vi của bảng không (1 <= j <= n)
        if (j <= n) count++;
    }
}


void output()
{
    freopen(output_file, "w", stdout);

    cout << count;
}


int main()
{
    input();
    process();
    output();

    return 0;
}